subsys/net/lib/dhcpv4/dhcpv4_server.c

```diff
@@ -498,6 +515,14 @@ static uint8_t *dhcpv4_encode_requested_params(
 				goto out;
 			}
 			break;
+
+		case DHCPV4_OPTIONS_DNS_SERVER:
+			buf = dhcpv4_encode_dns_server_option(
+				buf, buflen, &ctx->iface->config.ip.ipv4->gw);
```

Comment:
Why are we sending router address to client, shouldn't we send DNS server address instead?

Reply:
I might be wrong or lack some knowledge but I've often noticed that in a local network, the router's IP address is used as the DNS server address when a host is configured as a server that's why I used the router address as a DNS server address.

Reply:
It is true that in many cases the local router is also acting as a DNS server, but in that case the DNS server address in Zephyr is probably set to that address then.